### PR TITLE
Web-648: OcTableFiles enable multiple highlighting using selection

### PR DESCRIPTION
**Pull request by paulcod3**
## Description
See https://github.com/owncloud/web/issues/5164

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [ ] Tests

## Checklist:

- [x] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation added/updated

### CODE REVIEW

File: changelog/unreleased/change-enable-multiple-highlighting [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Change: Enable multiple highlighting for OcTableFiles
+
+We changed the highlighting in a way that now every selected file
+is highlighted.
```
Comment (kulmann) [inline changelog/unreleased/change-enable-multiple-highlighting:4]: Please add description about why this is a breaking change, e.g. the following:
```suggestion
is highlighted automatically. The `highlighted` prop has been removed as it's not used anymore. 
```

File: src/components/table/OcTableFiles.vue [hunk 2]
```diff
@@ -379,6 +371,12 @@ export default {
     areAllResourcesSelected() {
       return this.selection.length === this.resources.length
     },
+
+    selectedIds() {
+      let ids = []
```
Comment (kulmann) [inline src/components/table/OcTableFiles.vue:376]: Computed prop can be solved as a one liner:
`return this.selectedResources.map(r => r.id)`

**Review by kulmann — CHANGES_REQUESTED** [review]
Cool, didn't expect the changes to be so minimal. Awesome! Added some ideas for improvement.

File: src/components/table/OcTableFiles.vue (hunk 2 above)
Comment (paulcod3) [reply src/components/table/OcTableFiles.vue:376]: yes that looks better

**Comment by kulmann** [comment]
As discussed in a call:
- the `highlight` event from `OcTable` needs to be used to update the selection model. don't emit `showDetails` anymore.
- in oC Web we'll watch the selection model to decide whether or not the right sidebar needs to be shown
- the `show details` button needs a new implementation: stop the propagation of the click event and add a new method for it, which resets the selection model to the resource id of the clicked row/button (in oC Web this results in a right sidebar for the clicked row)

**Comment by paulcod3** [comment]
I would only change the unittests so they work for now since the highlighting logic will changed again soon.

**Review by kulmann — APPROVED** [review]
LGTM 👍

**Pull request merged**